### PR TITLE
Add SOURCE_SSL_VERIFY_SERVER_CERT parameter

### PR DESCRIPTION
**Pull request by aneustroev**
##### SUMMARY
Add SOURCE_SSL_VERIFY_SERVER_CERT, because it very important for security.

##### ISSUE TYPE
- Feature Pull Request

##### ADDITIONAL INFORMATION
This option help to protect from mitm attacks

### CODE REVIEW

File: plugins/modules/mysql_replication.py [hunk 1]
```diff
@@ -131,6 +131,11 @@
       L(MySQL encrypted replication documentation,https://dev.mysql.com/doc/refman/8.0/en/replication-solutions-encrypted-connections.html).
     type: str
     aliases: [master_ssl_cipher]
+  primary_ssl_verify_server_cert:
+    description:
+    - Same as mysql variable.
+    type: bool
+    aliases: [master_ssl_verify_server_cert]
```
Comment (Andersson007) [inline plugins/modules/mysql_replication.py]:
```suggestion
```
the other aliases exist for backwards compatibility, so no need to have it for new arguments

File: plugins/modules/mysql_replication.py [hunk 2]
```diff
@@ -458,6 +463,7 @@ def main():
         primary_ssl_cert=dict(type='str', aliases=['master_ssl_cert']),
         primary_ssl_key=dict(type='str', no_log=False, aliases=['master_ssl_key']),
         primary_ssl_cipher=dict(type='str', aliases=['master_ssl_cipher']),
+        primary_ssl_verify_server_cert=dict(type='bool', aliases=['master_ssl_verify_server_cert']),
```
Comment (Andersson007) [inline plugins/modules/mysql_replication.py]:
```suggestion
        primary_ssl_verify_server_cert=dict(type='bool', default=false),
```

File: plugins/modules/mysql_replication.py [hunk 3]
```diff
@@ -131,6 +131,11 @@
       L(MySQL encrypted replication documentation,https://dev.mysql.com/doc/refman/8.0/en/replication-solutions-encrypted-connections.html).
     type: str
     aliases: [master_ssl_cipher]
+  primary_ssl_verify_server_cert:
+    description:
+    - Same as mysql variable.
+    type: bool
```
Comment (Andersson007) [inline plugins/modules/mysql_replication.py:137]:
```suggestion
    type: bool
    default: false
    version_added: '3.4.5'
```

**Review by Andersson007 — COMMENTED** [review]
@aneustroev hello, thanks for the PR!
Could you please add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment)?

File: plugins/modules/mysql_replication.py (hunk 2 above)
Comment (aneustroev) [reply plugins/modules/mysql_replication.py]: Why i should remove alias?

Comment (aneustroev) [reply plugins/modules/mysql_replication.py]: oh, i understand

File: plugins/modules/mysql_replication.py [hunk 4]
```diff
@@ -131,6 +131,12 @@
...
+    default: false
+    version_added: '3.4.5'
```
Comment (laurent-indermuehle) [inline plugins/modules/mysql_replication.py]:
```suggestion
    version_added: '3.4.1'
```

**Review by laurent-indermuehle — DISMISSED** [review]

File: plugins/modules/mysql_replication.py (hunk 4 above)
Comment (Andersson007) [reply plugins/modules/mysql_replication.py]: sorry it should be `3.5.0`, it was a typo. We follow semver

Comment (Andersson007) [inline plugins/modules/mysql_replication.py]:
```suggestion
    version_added: '3.5.0'
```

Comment (Andersson007) [reply plugins/modules/mysql_replication.py]: sorry, i made a typo, we follow semver, this is a new feature, so the second number should be increased

**Comment by aneustroev** [comment]
Now is all right? When you can release version 3.5.0?

**Comment by laurent-indermuehle** [comment]
I've planned to do that by the end of septembre.

File: plugins/modules/mysql_replication.py [hunk 5]
```diff
@@ -595,6 +603,8 @@ def main():
             chm.append("MASTER_SSL_KEY='%s'" % primary_ssl_key)
         if primary_ssl_cipher is not None:
             chm.append("MASTER_SSL_CIPHER='%s'" % primary_ssl_cipher)
+        if primary_ssl_verify_server_cert:
```
Comment (laurent-indermuehle) [inline plugins/modules/mysql_replication.py:606]:
```suggestion
        if primary_ssl_verify_server_cert is not None:
```

File: plugins/modules/mysql_replication.py [hunk 6]
```diff
@@ -595,6 +603,8 @@ def main():
...
+            chm.append("SOURCE_SSL_VERIFY_SERVER_CERT=1")
```
Comment (laurent-indermuehle) [inline plugins/modules/mysql_replication.py:607]:
```suggestion
            chm.append("SOURCE_SSL_VERIFY_SERVER_CERT='%s'" % primary_ssl_verify_server_cert)
```

File: plugins/modules/mysql_replication.py (hunk 5 above)
Comment (laurent-indermuehle) [reply plugins/modules/mysql_replication.py:606]: Your version is simpler. But we lose in concistency.

**Review by laurent-indermuehle — CHANGES_REQUESTED** [review]

**Review by laurent-indermuehle — APPROVED** [review]
Sorry for my previous review.Plenty of other parameters uses the same syntax. So it look perfect to me.
Thanks for the new feature!

**Review by laurent-indermuehle — APPROVED** [review]
Sorry for my preview review that I have deleted just now. I hadn't seen the whole file and plenty of other parameters uses the same syntax.
Thanks for the new feature!

File: changelogs/fragments/435-mysql_replication_verify_server_cert.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - "mysql_replication - add new option: primary_ssl_verify_server_cert"
```
Comment (Andersson007) [inline changelogs/fragments/435-mysql_replication_verify_server_cert.yml]:
```suggestion
  - "mysql_replication - add a new option: ``primary_ssl_verify_server_cert`` (https://github.com/ansible-collections/community.mysql/pull/435)."
```

**Review by Andersson007 — COMMENTED** [review]
LGTM, could you please merge one formatting thing? then we'll merge the PR

**Comment by aneustroev** [comment]
It's my first PR, thank you for all your suggestions!)

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@aneustroev thanks for the contribution! Would be great to see more:)
@laurent-indermuehle thanks for reviewing!

**Pull request merged**